editor/store/selectors.js

```diff
@@ -1243,17 +1244,17 @@ function buildInserterItemFromBlockType( state, enabledBlockTypes, blockType ) {
  * Given a shared block, constructs an item that appears in the inserter.
  *
  * @param {Object}           state             Global application state.
- * @param {string[]|boolean} enabledBlockTypes Enabled block types, or true/false to enable/disable all types.
+ * @param {string[]|boolean} allowedBlockTypes Allowed block types, or true/false to enable/disable all types.
  * @param {Object}           sharedBlock       Shared block, likely from getSharedBlock().
  *
  * @return {Editor.InserterItem} Item that appears in inserter.
  */
-function buildInserterItemFromSharedBlock( state, enabledBlockTypes, sharedBlock ) {
-	if ( ! enabledBlockTypes || ! sharedBlock ) {
+function buildInserterItemFromSharedBlock( state, allowedBlockTypes, sharedBlock ) {
+	if ( ! allowedBlockTypes || ! sharedBlock ) {
 		return null;
 	}
 
-	const blockTypeIsDisabled = Array.isArray( enabledBlockTypes ) && ! includes( enabledBlockTypes, 'core/block' );
+	const blockTypeIsDisabled = Array.isArray( allowedBlockTypes ) && ! includes( allowedBlockTypes, 'core/block' );
```

Comment:
Unrelated to this PR, but I wanted to double check if we need to open another issue. Shouldn't this logic verify whether `core/block` isn't allowed, but also `referencedBlockType.name`?

Reply:
In my opinion, disabling `core/block` should prevent you from inserting _all_ shared blocks, but I could be persuaded otherwise. Let's discuss in https://github.com/WordPress/gutenberg/issues/5893.